### PR TITLE
Add educational institution status#363/zubkov

### PR DESCRIPTION
**Pull request by olegz7**

### CODE REVIEW

File: src/app/shared/models/provider.model.ts [hunk 1]
```diff
@@ -43,6 +44,7 @@ export class Provider {
     this.actualAddress = actualAddress;
     this.description = photo.description;
     this.userId = user.id;
+    this.institutionStatusId = photo.institutionStatusId? photo.institutionStatusId : null
```
[the remark this answers is not in the document]
Comment (dmitryy90) [reply src/app/shared/models/provider.model.ts:47]: I believe it is not photo anymore, lets rename photo parameter

File: src/app/shared/services/provider/provider.service.ts [hunk 2]
```diff
@@ -40,4 +41,11 @@ export class ProviderService {
   updateProvider(provider: Provider): Observable<object> {
     return this.http.put('/api/v1/Provider/Update', provider);
   }
+
+  /**
+ * This method get all institution statuses
+ */
+    getInstitutionStatus(): Observable<InstitutionStatus[]> {
+    return this.http.get<InstitutionStatus[]>('/api/v1/InstitutionStatus/Get');
```
Comment (dmitryy90) [inline src/app/shared/services/provider/provider.service.ts:49]: indentations

File: src/app/shell/personal-cabinet/provider/provider-org-info/provider-org-info.component.ts [hunk 3]
```diff
@@ -36,4 +61,7 @@ export class ProviderOrgInfoComponent implements OnInit {
     this.editLink = createProviderSteps[tabChangeEvent.index];
   }
 
+
```
Comment (dmitryy90) [inline src/app/shell/personal-cabinet/provider/provider-org-info/provider-org-info.component.ts:64]: ?

**Review by dmitryy90 — CHANGES_REQUESTED** [review]

File: src/app/shell/personal-cabinet/provider/create-provider/create-provider.guard.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export class CreateProviderGuard implements CanDeactivate<unknown>, CanLoad {
     segments: UrlSegment[]): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
     const isEditMode = this.store.selectSnapshot<boolean>(AppState.isEditMode);
 
-    return isEditMode ? true : this.user$.pipe(map((user: User) => user.role === Role.provider && user.isRegistered === false));
+    return isEditMode ? true : this.user$.pipe(map((user: User) => user?.role === Role.provider && user.isRegistered === false));
```
Comment (dmitryy90) [inline src/app/shell/personal-cabinet/provider/create-provider/create-provider.guard.ts:29]: in which cases user can be undefined?

**Review by dmitryy90 — CHANGES_REQUESTED** [review]

File: src/app/shell/personal-cabinet/provider/create-provider/create-provider.guard.ts (hunk 4 above)
Comment (olegz7) [reply src/app/shell/personal-cabinet/provider/create-provider/create-provider.guard.ts]: thanks for catching that, I think it was for testing purposes, deleted '?'

**Review by dmitryy90 — APPROVED** [review]

**Pull request merged**